core/trino-main/src/main/java/io/trino/server/PluginManager.java

```diff
@@ -70,6 +72,7 @@
     private final AccessControlManager accessControlManager;
     private final Optional<PasswordAuthenticatorManager> passwordAuthenticatorManager;
     private final CertificateAuthenticatorManager certificateAuthenticatorManager;
+    private final Optional<HeaderBasedAuthenticatorManager> headerBasedAuthenticatorManager;
```

Comment:
Why is this optional?

Reply:
to tell you the truth, I am at quite a loss as to what's happening here. I just followed how the password manager is adding authenticators. I initially did not use an optional but then saw this error 
<img width="935" alt="Screen Shot 2021-08-30 at 5 16 50 PM" src="https://user-images.githubusercontent.com/19635283/131422433-96d49054-86fb-4825-bbb4-8f9210ec78bd.png">
Having never worked with guice, I'm not sure how to interpret this, please let me know if this is due to an implementation flaw.

Reply:
Test setup things differently than the product. It is a bit unfortunate, but it looks like you need to also do guice binding for query runner too.

Reply:
Hey @kokosing, sorry I'm not quite following, you can chalk it up to inexperience, could you please elaborate.

Reply:
I believe @kokosing's question is higher level: why are we binding and using this as optional everywhere?

The simple answer is because that's what `PasswordAuthenticatorManager` works. But why? It was done in #7151 so that `PasswordAuthenticatorConfig` is only read if password authentication is enabled. For example, this means that setting `password-authenticator.config-files` is an error if you aren't using password authentication.

This results in stricter config validation, which is the behavior we want and we try to use this pattern as much as possible. For example, we recently split out `HttpsConfig` for the HTTP server, so that you are only allowed to set `http-server.https.*` properties if HTTPS is enabled, which prevents mistakes.

The way all the authenticator stuff gets wired up is a bit confusing and convoluted, with the `Manager` and the `Authenticator` classes and the `setRequired()` and optional bindings, but it works, so copying the pattern here and being consistent is best. We can revisit the entire design later if needed or if someone has a better idea on how to structure it.